lang/cs/Org.Apache.REEF.Network.Tests/NamingService/NameServerTests.cs

```diff
@@ -233,6 +234,33 @@ public void TestConstructorInjection()
             }
         }
 
+        [TestMethod]
+        public void TestNameCache()
+        {
+            double interval = 1000;
+            var config =
+                TangFactory.GetTang()
+                    .NewConfigurationBuilder()
+                    .BindNamedParameter<NameCacheConfiguration.CacheEntryExpiryTime, double>(
+                        GenericType<NameCacheConfiguration.CacheEntryExpiryTime>.Class,
+                        interval.ToString(CultureInfo.InvariantCulture))
+                    .Build();
+            
+            var injector = TangFactory.GetTang().NewInjector(config);
+            var cache = injector.GetInstance<NameCache>();
+
+            cache.Set("dst1", new IPEndPoint(IPAddress.Any, 0));
+            Thread.Sleep(2000);
```

Comment:
This is to test whether expiry works in NameCache or not. If we do not want
to test this, I can remove the test.

Dhruv

On Mon, Jul 27, 2015 at 10:59 AM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Network.Tests/NamingService/NameServerTests.cs
> <https://github.com/apache/incubator-reef/pull/318#discussion_r35565839>:
>
> > +        public void TestNameCache()
> > +        {
> > +            double interval = 1000;
> > +            var config =
> > +                TangFactory.GetTang()
> > +                    .NewConfigurationBuilder()
> > +                    .BindNamedParameter<NameCacheConfiguration.CacheEntryExpiryTime, double>(
> > +                        GenericType<NameCacheConfiguration.CacheEntryExpiryTime>.Class,
> > +                        interval.ToString(CultureInfo.InvariantCulture))
> > +                    .Build();
> > +
> > +            var injector = TangFactory.GetTang().NewInjector(config);
> > +            var cache = injector.GetInstance<NameCache>();
> > +
> > +            cache.Set("dst1", new IPEndPoint(IPAddress.Any, 0));
> > +            Thread.Sleep(2000);
>
> Do we have to do this? Real time units are dangerous in tests ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/318/files#r35565839>.
>